Kernel/Arch/aarch64/Time/PL031.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025, the SerenityOS developers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <AK/NeverDestroyed.h>
+#include <AK/Optional.h>
+#include <Kernel/Arch/aarch64/Time/PL031.h>
+#include <Kernel/Firmware/DeviceTree/Driver.h>
+#include <Kernel/Firmware/DeviceTree/Management.h>
+
+namespace Kernel {
+
+static NeverDestroyed<Optional<PL031>> s_rtc;
+static NeverDestroyed<Mutex> s_rtc_lock { "PL031"sv };
+
+PL031::PL031(Memory::TypedMapping<RTCRegisters volatile> rtc_registers)
+    : m_rtc_registers(move(rtc_registers))
+{
+    m_boot_time = m_rtc_registers->RTCDR;
```

Comment:
I can't find anything in the datasheet saying that the data register is supposed to contain the number of seconds since the unix epoch, only that it increments by 1 tick per second.

(Also, nice that this peripheral is affected by the Y2038 problem.)

Reply:
The x86 RTC gives you the time in year,month,day,hour,minute,second format.
But it seems like both qemu and linux treat it as a unix timestamp, so it's fine I guess.

I guess it's up to the user of the RTC to decide the epoch. And if we would support setting the RTC we would also use the unix epoch.